### DISCUSSION TITLE
Use OpenModelica to open a Dymola based package

### DISCUSSION THREAD

**Discussion by yifanm4** [OpenModelica/OpenModelica#12722] [Q&A]
Hello everyone, 

I am interested in using the TRANSFORM-library (https://github.com/ORNL-Modelica/TRANSFORM-Library) in my OpenModelica environment for research purpose. TRANSFORM was originally designed in Dymola. 

I am pretty new to Modelica environment, so I am not sure how difficult it is for a translation between these two languages (If it's even possible) .  

I started from running a pipe example from TRANSFORM.Fluid.Pipes.Examples.RELAPComparison.HeatedPipe_SteadyState and get the following error messages: 

`[8]14:39:33 Translation Error
[CodegenCFunctions.tpl: 7169:13-7169:13]: Template error: ASUB non-scalar TRANSFORM.Fluid.Pipes.Examples.RELAPComparison.HeatedPipe_SteadyState.pipe.colorMap(pipe.n_colors)[1 + integer(0.7875000000000001 * (max(293.15, min(pipe.val, 373.15)) - 293.15))]. The inner exp has type: Real[64, 3]. After ASUB it is still an array: Real[3].` 

` [9] 14:39:33 Translation Error
[CodegenCFunctions.tpl: 7169:13-7169:13]: Template error: ASUB non-scalar TRANSFORM.Fluid.Pipes.Examples.RELAPComparison.HeatedPipe_SteadyState.pipe.colorMap(pipe.n_colors)[1 + integer(0.7875000000000001 * (max(293.15, min(pipe.val, 373.15)) - 293.15), 0)]. The inner exp has type: Real[64, 3]. After ASUB it is still an array: Real[3]..` 

` [10] 14:39:33 Translation Error
[C:/OM122/OM64bit/OMCompiler/Compiler/SimCode/SimCodeMain.mo: 532:7-532:78]: Internal error C:\Program Files\OpenModelica1.22.3-64bit\bin\OMEdit.exe: [unknown function name] failed`

**Comment by yifanm4** [comment]
FYI, I asked the question from the TRANSFORM side, here is what I get: 

> Good day. There was some testing a while back but found OpenModelica not quite capable enough. Recently OM has made some improvements that, couple with some minor modifications to a few TRANSFORM components, there may be a path forward in using OM for at least some tests. We do have a little bit of effort looking at OM currently that may make some improvements but I don't have any other information on that at the moment.

> So in summary, you can try and may find some success however we don't have any recent first hand success using OM we can immediately share. Dymola remains the recommended tool for using TRANSFORM if possible (there are academic rates I believe).

**Comment by casella** [comment]
I would suggest to check the User's Guide, [Porting Modelica Libraries to OpenModelica](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/porting.html). Feel free to ask further questions 😃
